src/lib/message/L1ToL2Message.ts

```diff
@@ -259,6 +283,68 @@ export class L1ToL2MessageReader extends L1ToL2Message {
     return null
   }
 
+  private async getFirstRedeem(
+    retryableCreationReceipt?: TransactionReceipt,
+    includeFail?: boolean
+  ): Promise<TransactionReceipt | null> {
+    includeFail = includeFail || false
+    const l2Network = await getL2Network(this.l2Provider)
+    const creationReceipt =
+      retryableCreationReceipt ||
+      (await this.l2Provider.getTransactionReceipt(this.retryableCreationId))
```

Comment:
Instead of potentially looking up again, lets make `getRetryableCreationReceipt` cache the results of trying to fetch the receipt

Reply:
yea agreed this will prob be cleaner